docs/devel/hello-world-gadget.md

```diff
@@ -550,6 +550,127 @@ RUNTIME.CONTAINERNAME  PID          COMM         FILENAME
 
 Now the UID and GID fields are also printed.
 
+## Adding test file for the gadget
+
+Need to verify that gadget works as you expected it to?
+Let's create a test file called `mygadget_test.go` and put the following contents in there.
+
+First we would need to import some packages.
+
+```golang
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	// helper functions for creating and running commands in a container.
+	"github.com/inspektor-gadget/inspektor-gadget/integration"
+
+  	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/command"
+
+  	// wrapper function for ig binary
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+
+	// helper functions for parsing and comparing output.
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+
+  	// Event struct for fields enriched by ig.
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+```
+
+Then, we create a structure with all the information the gadget provides.
+
+```golang
+type mygadgetEvent struct {
+	eventtypes.Event
+
+	MountNsID uint64 `json:"mountnsid"`
+	Pid       uint32 `json:"pid"`
+	Uid       uint32 `json:"uid"`
+	Gid       uint32 `json:"gid"`
+	Comm      string `json:"comm"`
+	Filename  string `json:"filename"`
+}
+```
+
+Later we create a test function called `TestMyGadget()`. 
+In this, we first create a container manager (can be either `docker` or `containerd`). After that, we create a command to run the gadget with various options. 
+Finally, we pass the commands for running the gadget and container command to `RunTestSteps()` which executes them in an order.
+
+```golang
+func TestMyGadget(t *testing.T) {
+	cn := "test-mygadget"
+
+  	// returns a container manager which implements an interface with methods for creating new container
+  	// and running commands within that container.
+	containerFactory, err := integration.NewContainerFactory("docker")
+	require.NoError(t, err, "new container factory")
+
+	mygadgetCmd := igrunner.New(
+    		// gadget repository and tag are added via env vars
```

Comment:
Actually, this comment is not up-to-date anymore as mentioned in https://github.com/inspektor-gadget/inspektor-gadget/pull/2607#discussion_r1548088438.

Reply:
Should I remove this whole comment and explain that gadget repo and tag are added automatically by IG on this [line](https://github.com/inspektor-gadget/inspektor-gadget/pull/2607/files#diff-5707bfb4f5107fcae74a123973315fe44d85c9b92215454ecfed0085ddc669f5R661)?

Reply:
Since IG and not ig wrapper adds repo and tag to the gadget name, this difference does not affect the users so, we should still let them know that these variables need to be defined,